tests/src/Integration/Action/UserUnblockTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\UserUnBlockTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\UserUnBlock
+ * @group rules_actions
+ */
+class UserUnblockTest extends RulesEntityIntegrationTestBase {
+
+  /**
+   * Constant used for authenticated test when mocking a user.
+   */
+  const AUTHENTICATED = TRUE;
+
+  /**
+   * Constant used for authenticated test when mocking a user.
+   */
+  const ANONYMOUS = FALSE;
+
+  /**
+   * Constant used for active test when mocking a user.
+   */
+  const ACTIVE = TRUE;
+
+  /**
+   * Constant used for active test when mocking a user.
+   */
+  const BLOCKED = FALSE;
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Engine\RulesActionInterface
+   */
+  protected $action;
+
+
+  /**
+   * @dataProvider userProvider
+   * @covers ::execute
+   */
+  public function testUnblockUser($active, $authenticated, $expects) {
+    $user = $this->getMock('Drupal\user\UserInterface');
+
+    $user = $this->getMock('Drupal\user\UserInterface');
+
+    $user->expects($this->any())
+      ->method('isBlocked')
+      ->willReturn(!$active);
+
+
+    $user->expects($this->any())
```

Comment:
once makes more sense here as well. any means 0 or more. let's be a bit more strict